### PR TITLE
Issue 1235 dataset

### PR DESCRIPTION
**Pull request by zebanaqviWAT**

### CODE REVIEW

File: pyaedt/application/Design.py [hunk 1]
```diff
@@ -2242,13 +2242,15 @@ def import_dataset1d(self, filename, dsname=None, is_project_dataset=True):
         )
 
     @pyaedt_function_handler()
-    def import_dataset3d(self, filename, dsname=None):
+    def import_dataset3d(self, filename, encoding="utf-8-sig", dsname=None):
```
Comment (maxcapodi78) [inline pyaedt/application/Design.py:2245]: To keep back compatibility better to add the new args at the end of the method

```suggestion
    def import_dataset3d(self, filename, dsname=None, encoding="utf-8-sig"):
```

**Review by maxcapodi78 — APPROVED** [review]
LGTM

**Review by maxcapodi78 — CHANGES_REQUESTED** [review]
apart a small suggestion it looks good to me

File: _unittest/test_01_Design.py [hunk 2]
```diff
@@ -204,14 +204,21 @@ def test_19_import_dataset1d(self):
...
+        ds8 = self.aedtapp.import_dataset3d(filename, dsname="dataset_csv")
+        assert ds8.name == "$dataset_csv"
+        assert ds8.delete()
         ds10 = self.aedtapp.import_dataset3d(filename, dsname="$dataset_test")
-        assert ds10.vunit == "m_per_sec"
+        assert ds10.zunit == "mm"
+        filename = os.path.join(local_path, "example_models", "Dataset_3D.csv")
+        ds8 = self.aedtapp.import_dataset3d(filename, encoding="utf-8-sig", dsname="dataset_csv")
+        assert ds8.name == "$dataset_csv"
+        filename = os.path.join(local_path, "example_models", "Dataset_3D.xlsx")
```
Comment (maxcapodi78) [inline _unittest/test_01_Design.py:219]: To test the excel import you need pandas on the build machine. add it to requirements_test.txt

**Review by maxcapodi78 — CHANGES_REQUESTED** [review]
2 changes needed

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM

**Review by Samuelopez-ansys — APPROVED** [review]

File: pyaedt/application/Design.py [hunk 3]
```diff
@@ -2261,12 +2263,58 @@ def import_dataset3d(self, filename, dsname=None):
...
+        index_of_dot = filename.rfind(".")
+        file_extension = filename[index_of_dot + 1 :]
+        xlist = []
+        ylist = []
+        zlist = []
+        vlist = []
 
-        header_list = header.split("\t")
+        if file_extension == "xlsx":
+            self.logger.warning("Warning: You need pandas installed for reading excel files")
```
Comment (MaxJPRey) [inline pyaedt/application/Design.py:2274]:
```suggestion
            self.logger.warning("You need pandas library installed for reading excel files")
```

File: pyaedt/application/Design.py [hunk 4]
```diff
@@ -2261,12 +2263,58 @@ def import_dataset3d(self, filename, dsname=None):
...
+            try:
+                import pandas as pd
+            except ImportError:
+                self.logger.error(
+                    "Pandas is not installed. Either install pandas or save the file as .csv or .tab. " "Exiting."
```
Comment (MaxJPRey) [inline pyaedt/application/Design.py:2279]: What is the `" "` for before `Exiting` in the log message?

File: pyaedt/application/Design.py [hunk 5]
```diff
@@ -2261,12 +2263,58 @@ def import_dataset3d(self, filename, dsname=None):
...
+                )
+                return
+
+            df = pd.read_excel(filename)
```
Comment (MaxJPRey) [inline pyaedt/application/Design.py:2283]: If possible, could you use a more specific variable name?

File: pyaedt/application/Design.py [hunk 6]
```diff
@@ -2261,12 +2263,58 @@ def import_dataset3d(self, filename, dsname=None):
...
+        elif file_extension == "tab":
+            with open(filename) as my_file:
+                lines = my_file.readlines()
+            header = lines[0]
+            for item in lines[1:]:
+                xlist.append(float(item.split()[0]))
+                ylist.append(float(item.split()[1]))
+                zlist.append(float(item.split()[2]))
+                vlist.append(float(item.split()[3]))
+            my_file.close()
```
Comment (MaxJPRey) [inline pyaedt/application/Design.py:2316]: @zebanaqviWAT You don't need to close the file hear as you are using a `with open` statement that will take care of closing the file after reading its content (here with the `readlines()` method).
```suggestion
```

**Review by maxcapodi78 — APPROVED** [review]
lgtm

**Pull request merged**